src/main/java/org/folio/circulation/services/EventPublisher.java

```diff
@@ -198,9 +225,13 @@ public EventPublisher(PubSubPublishingService pubSubPublishingService) {
         runAsync(() -> publishRenewedEvent(loan.copy().withUser(user)));
       }
 
-      return pubSubPublishingService.publishEvent(LOAN_DUE_DATE_CHANGED.name(),
-        payloadJsonObject.encode())
-        .thenApply(r -> succeeded(loan));
+      return pubSubPublishingService.publishEvent(LOAN_DUE_DATE_CHANGED.name(), payloadJsonObject.encode())
+        .handle((result, error) -> {
+          if (error != null) {
+            return failedDueToServerError(error.getMessage());
+          }
+          return succeeded(loan);
+        });
```

Comment:
Could you please try to extract this code, which is common for all methods that return `Result<Loan>,` into separate private method?

Reply:
I think, this not good idea, since some of method return different result, but few of them returns same result

Reply:
fixed